File: app/src/main/java/io/github/gmathi/novellibrary/model/NovelGenre.java
package io.github.gmathi.novellibrary.model;

public class NovelGenre {
    private long novelId;
    private long genreId;

    public NovelGenre() {

    }

    public NovelGenre(long novelId, long genreId) {
        this.novelId = novelId;
        this.genreId = genreId;
    }
    
    public long getNovelId() {
        return this.novelId;
    }
    
    public void setNovelId(long novelId) {
        this.novelId = novelId;
    }
    
    public long getGenreId() {
        return this.genreId;
    }
    
    public void setGenreId(long genreId) {
        this.genreId = genreId;
    }
    
}

File: app/src/main/java/io/github/gmathi/novellibrary/model/database/NovelGenre.kt
package io.github.gmathi.novellibrary.model.database

import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity(tableName = "novel_genre", primaryKeys = ["novel_id", "genre_id"])
data class NovelGenre(
    @ColumnInfo(name = "novel_id") var novelId: Long = 0,
    @ColumnInfo(name = "genre_id") var genreId: Long = 0
)
